### DISCUSSION TITLE
Ioc Error after upgrade adonis v5 to in-memory compilation

### DISCUSSION THREAD

**Discussion by mlusca** [adonisjs/core#1803] [Help]
Before updating adonis, the IoC was functioning normally.
Going back to the commit before the upgrade is working, and after the upgrade it doesn't work anymore.

Upgrade guide: https://preview.adonisjs.com/blog/october-2020-release

Error:
"E_IOC_LOOKUP_FAILED: Cannot resolve \"Adonis/Lucid/Relations\" namespace from the IoC Container",

AppProvider:

```
 public register() {
    this.$container.singleton('Mota/Auth', () => {
      const database = this.$container.use('Adonis/Lucid/Database')
      const service = this.$container.use('App/Modules/Auth/Services/UsuarioService')
      const usuarioService = new service.default()

      return new AuthenticateService(database, usuarioService)
    })
  }
```
Commenting on the relationship in the User model works:
```
  // @hasMany(() => UsuariosToken)
  // public usuariosToken: HasMany<typeof UsuariosToken>

```

**Comment by thetutlage** [comment]
Where are you using `Mota/Auth` namespace? The error will appear when you are trying to use IoC container bindings inside the config files, since the container is not ready at that time

**Reply by mlusca** [reply]
@thetutlage Hi, it wasn't really that. I was importing HasMany from Adonis / Lucid / Relations and that was what was causing the problem, I changed the import to Adonis / Lucid / Orm and the error stopped. I don't know if this should really happen, but correct it like that.
